### ISSUE TITLE
About plans of ceph-cosi project

### ISSUE THREAD

**Issue by cyb70289** [ceph/ceph-cosi#4]
Hi,

I'm interested in `ceph + cosi` and glad to find this exciting repo.

Looks there are many todos for current ceph-cosi project.
Do we have plans to promote ceph-cosi as a product, or we'd rather keep it simple as a reference?

I'm glad to help if we want to improve ceph-cosi project.
I suppose the first step is to setup CI (current code even fails building).

Yibo

**Comment by thotz** [comment]
@cyb70289 Thanks for reaching out, currently COSI API undergoing a few design changes[1]. I waiting for that to be finalised so that I can refresh this repo.  Hopefully, by end of this month, everything will be sorted out.
The plan is to promote ceph-cosi as a product than a simple reference. And by k8s 1.23 hopefully we will get alpha version of both COSI and Ceph-cosi driver

[1] https://github.com/kubernetes/enhancements/pull/2813/

**Comment by thotz** [comment]
@cyb70289 Yeah that's the plan, I have just started intial PR[1] using GitHub actions https://github.com/ceph/ceph-cosi/pull/3 bit back. But didn't rework on it. One of the issues with the current approach is how to RGW server. I don't want to use Rook as ceph-csi does but the plan to use `cephadm` locally the script was working fine and I didn't look deep into in the github action failure. Another option for setting up rgw and ceph cluster is how go-ceph is setting up. The more elegant way might be cephadm(even rgwadm is also planned as well), but for the time being, we can rely on go-ceph script[3]. 

[1] https://github.com/ceph/ceph-cosi/pull/3
[2] https://github.com/ceph/ceph/pull/42710/
[3] https://github.com/ceph/go-ceph/blob/master/micro-osd.sh
